utils/fdb_wrapper_cpp/include/fdbcs.hpp

```diff
@@ -104,5 +110,76 @@ class DataBaseCreator
   static std::shared_ptr<FDBDataBase> createDataBase(const std::string clusterFilePath);
 };
 
+using Block = std::pair<uint32_t, std::string>;
+using Key = std::string;
+using Keys = std::vector<Key>;
+
+class KeyGenerator
+{
+ public:
+  virtual ~KeyGenerator()
+  {
+  }
+  virtual Key generateKey() = 0;
+  virtual uint32_t getKeySize() = 0;
+};
+
+class BoostUIDKeyGenerator : public KeyGenerator
+{
+ public:
+  Key generateKey() override;
+  uint32_t getKeySize() override;
+};
+
+// This class represetns a machinery to handle a data `blob`.
+class BlobHandler
+{
+ public:
+  BlobHandler(std::shared_ptr<KeyGenerator> keyGen, uint32_t blockSizeInBytes = 100000)
+   : keyGen_(keyGen), blockSizeInBytes_(blockSizeInBytes)
+  {
+    // Block size in 100KB shows the best performance.
+    keySizeInBytes_ = keyGen_->getKeySize();
+    assert(keySizeInBytes_);
```

Comment:
The asserts are necessary here?
They will cause SIGABRT if they are  not caught properly thus SM will crash.

Reply:
Is there other way to stop an execution with an error w/o using assert?

Reply:
AFAIK NDEBUG is set in the context of SM.

Reply:
The assert is using here is to prevent situation when we can choose a key and datablock size which could cause undefined behavior - division by zero and so on. Those params are not configurable during runtime, so it's only a compile time options and only for debug build.